app/scripts/controllers/transactions/tx-state-manager.test.js

```diff
@@ -614,6 +631,76 @@ describe('TransactionStateManager', function () {
       );
       assert.equal(result[1].id, 2, 'early txs truncated');
     });
+
+    it('cuts off entire groups of transactions by nonce when adding new transaction', function () {
+      const limit = txStateManager.txHistoryLimit;
+      // In this test case the earliest two transactions are a dropped attempted ether send and a
+      // following cancel transaction with the same nonce. these two transactions should be dropped
+      // together as soon as the 11th unique nonce is attempted to be added, mocked here with limit + 1
+      const txs = generateTransactions(limit + 1, {
+        chainId: currentChainId,
```

Comment:
should the `nonce` method param be defined here, to ensure more than once transaction of the same nonce is in the test?

Reply:
Yes! must have dropped during code cleanup

Reply:
fixed